src/layers/legacy/medVtkInria/vtkImageView/vtkImageView3D.cxx

```diff
@@ -505,26 +509,14 @@ void vtkImageView3D::InternalUpdate()
     bool multichannelInput = (this->m_poInternalImageFromInput->GetScalarType() == VTK_UNSIGNED_CHAR &&
                               (this->m_poInternalImageFromInput->GetNumberOfScalarComponents() == 3 ||
                                this->m_poInternalImageFromInput->GetNumberOfScalarComponents() == 4 ));
-    if(this->GetMedVtkImageInfo() == nullptr || !this->GetMedVtkImageInfo()->initialized)
-    {
-        this->Renderer->RemoveAllViewProps();
-        //TODO apparently RemoveAllViewProps() is not enough, though it should be
-        this->ActorX->GetMapper()->SetInputConnection(nullptr);
-        this->ActorY->GetMapper()->SetInputConnection(nullptr);
-        this->ActorZ->GetMapper()->SetInputConnection(nullptr);
-        this->Render();
-        return;
-    }
+
     vtkImageAppendComponents *appender = vtkImageAppendComponents::New();
     if (this->LayerInfoVec.size()>0 && !multichannelInput)
     {
         // append all scalar buffer into the same image
-        for( LayerInfoVecType::const_iterator it = this->LayerInfoVec.begin();
-             it!=this->LayerInfoVec.end(); ++it)
+        for (auto it : this->LayerInfoVec)
         {
-            if (!it->ImageDisplay->GetMedVtkImageInfo() || !it->ImageDisplay->GetMedVtkImageInfo()->initialized)
-                continue;
-            appender->AddInputConnection(it->ImageDisplay->GetInputProducer()->GetOutputPort());
+            appender->AddInputConnection(it.ImageDisplay->GetInputProducer()->GetOutputPort());
```

Comment:
. vs -> ???

Reply:
ha non c'est un "for:"

Reply:
c'était cette ligne (le "->") qui n'était pas trouvé et qui posait les problèmes de layers en 3d